walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,9 +14,44 @@
 
 package com.google.sps;
 
+import com.google.sps.TimeRange;
 import java.util.Collection;
 
 public final class FindMeetingQuery {
+  /**
+    Determines the free spaces within the working day not occupied by attended events
+    An event is considered attended if isAttended(event,request) returns true.
+    // Attended Event:     |--B--|
+    // Day     : |---------------------|
+    // Output : |----------1111111----|
+
+    @param Event[] events the (potentially unattended) events during the day
+    @param MeetingRequest request the meeting request used to determine event attendence.
+    @returns boolean[48] representing the day in 30 minute intervals, with occupied slots filled with a non-zero value (1)
+  */
+  private static boolean[] getFreeTimes(Event[] events, MeetingRequest request) {
+    final static int slots = 24 * 2; // 24 * 2 is the number of half hour slots in a day
+    final static int THIRTY_MINUTES = 30; //The divisor used to split up the day
+    
+    boolean[] freeTimes = new boolean[slots]; // Output array of free times in the day.
+
+    // Loop through events and mark slots as occupied if the event is attended.
+    for(int i = 0  i < events.length  i ++) {
+      // If an event is not attended, it does not need to be considered.
+      if(!isAttended(events[i], request)) continue;
+
+      TimeRange eventTime = events[i].getWhen();
+      int eventStart = eventTime.start() / THIRTY_MINUTES;
+      int eventEnd = (eventTime.start() + eventTime.duration()) / THIRTY_MINUTES - 1;
+      int eventDuration = eventTime.duration() / THIRTY_MINUTES;
```

Comment:
this is an unused variable. you can remove it.

Reply:
Removed in commit fedb6ad